onnxruntime/core/session/custom_ops.cc

```diff
@@ -422,117 +424,270 @@ common::Status CreateCustomRegistry(gsl::span<OrtCustomOpDomain* const> op_domai
     constexpr uint32_t min_ort_version_with_optional_io_support = 8;
     constexpr uint32_t min_ort_version_with_variadic_io_support = 14;
 
-    std::vector<ONNX_NAMESPACE::OpSchema> schemas_list;
     for (const auto* op : domain->custom_ops_) {
-      ONNX_NAMESPACE::OpSchema schema(op->GetName(op), "custom op registered at runtime", 0);
-
-      size_t type_id_counter = 0;
       const size_t input_count = op->GetInputTypeCount(op);
-      for (size_t i = 0; i < input_count; i++) {
-        onnx::OpSchema::FormalParameterOption option = onnx::OpSchema::FormalParameterOption::Single;
-        bool is_homogeneous = true;
-        int min_arity = 1;
+      const size_t output_count = op->GetOutputTypeCount(op);
+      int undefined = 0;
+      auto schema_map_iter = schema_map.find(op->GetName(op));
+      if (schema_map_iter == schema_map.end()) {
```

Comment:
not fully get the point, but i am wondering whether we make it too complicated. Here is my thoughts:
1. we build kernel definition for each custom_op instance first. the kernel definition will contains the type information for every argument.
2. when build schema, the type/shape inference function will capture the list of kernel definition on this op. we iterate each kernel definition to see which one match the input types, and use that kernel definition to infer output types.
3. we probably need a check to avoid register duplicated kernel with same types.

Will that simplify the algorithm here?